src/test/scala/mesosphere/marathon/core/matcher/manager/impl/OfferMatcherManagerActorTest.scala

```diff
@@ -195,6 +195,36 @@ class OfferMatcherManagerActorTest extends AkkaUnitTest with Eventually {
       Then("offer-1 is declined, since the actor did not respond in time")
       offerMatch1.future.futureValue.opsWithSource should be('empty)
     }
+
+    "not send offers to matchers that are not interested" in new Fixture {
+      Given("OfferMatcher not interested in offer")
+      val offer1 = offer()
+      val offerMatch1 = Promise[OfferMatcher.MatchedInstanceOps]
+      offerMatcherManager.underlyingActor.launchTokens = 100
+      val matcherMock = matcher(isInterestedIn = false)
+      offerMatcherManager.underlyingActor.matchers += matcherMock
+
+      When("Offer is sent to MatcherManager")
+      offerMatcherManager ! ActorOfferMatcher.MatchOffer(offer1, offerMatch1)
+
+      Then("OfferMatcher not interested in offer should not receive any offer")
+      verify(matcherMock, times(0)).matchOffer(any)
```

Comment:
Isn't this flaky? There is no guarantee that the message arrived when we do the check.

Reply:
good point, I changed those...